### PR TITLE
MEN-7029: Download log button failing to download file

### PR DESCRIPTION
**Pull request by mineralsfree**
Previous implementation didn't send authorization header 

Ticket: MEN-7029
Changelog: None

### CODE REVIEW

**Comment by aleksandrychev** [comment]
@mineralsfree this change doesn't look like a proper fix. what's wrong here, when you click a log link then API endpoint authorization does not happen?

**Comment by mineralsfree** [comment]
> @mineralsfree this change doesn't look like a proper fix. what's wrong here, when you click the log link then API endpoint authorization does not happen?

It opens a link in a new tab trying to fetch a file, but since JWT token is not passed in HTTP header, the request is not authorized

**Comment by aleksandrychev** [comment]
> > @mineralsfree this change doesn't look like a proper fix. what's wrong here, when you click the log link then API endpoint authorization does not happen?
> 
> It opens a link in a new tab trying to fetch a file, but since JWT token is not passed in HTTP header, the request is not authorized

JWT also set in cookies, and API in this case should also recognise it. @tranchitella @alfrunes please confirm

**Comment by alfrunes** [comment]
> JWT also set in cookies, and API in this case should also recognise it. @tranchitella @alfrunes please confirm

Correct, the backend uses the `JWT` cookie as the session token.

**Comment by mineralsfree** [comment]
Then it seems the problem is that the cookie is not saved/sent on frontend, I cannot see anything similar to JWT or other auth cookie in dev tools after I log in. @aleksandrychev @alfrunes @tranchitella

**Comment by aleksandrychev** [comment]
yes, right. now UI moves token from cookies to localStorage.  @mineralsfree you can use 
```cookies.set('JWT', token, { path: '/', secure: true, sameSite: 'strict', maxAge: 5 });``` before downloading, as we do for other downloads. maxAge will set cookies life time for a 5 seconds.

File: src/js/components/devices/device-details/deployments.js [hunk 1]
```diff
@@ -49,34 +53,58 @@ const EmptyState = ({ isFiltered }) => (
...
-        <Button component="a" href={`${window.location.origin}${deploymentsApiUrl}/deployments/${id}/devices/${deviceId}/log`} target="_blank">
-          Log
-        </Button>
-      )
-  }
-];
-
 const History = ({ className, items, page, perPage, setPage, setPerPage, total }) => {
+  const { token } = useSelector(getCurrentSession);
+  const columns = [
```
Comment (aleksandrychev) [inline src/js/components/devices/device-details/deployments.js:58]: no need to move columns inside the history component, you can pass token to the component

```
{columns.map(({ key, Component }) => (
                <TableCell key={`${item.id}-${key}`}>
                  <Component deviceDeployment={item} token={token} />
                </TableCell>
              ))}
```

and get it later 
```Component: ({ deviceDeployment: { id, log, deviceId }, token }) =>```

File: src/js/components/devices/device-details/deployments.js [hunk 2]
```diff
@@ -49,34 +53,58 @@ const EmptyState = ({ isFiltered }) => (
...
-      log && (
-        <Button component="a" href={`${window.location.origin}${deploymentsApiUrl}/deployments/${id}/devices/${deviceId}/log`} target="_blank">
-          Log
-        </Button>
-      )
-  }
-];
-
 const History = ({ className, items, page, perPage, setPage, setPerPage, total }) => {
+  const { token } = useSelector(getCurrentSession);
```
Comment (aleksandrychev) [inline src/js/components/devices/device-details/deployments.js:57]: please use getToken() from auth

File: src/js/components/devices/device-details/deployments.js [hunk 3]
```diff
@@ -49,34 +53,58 @@ const EmptyState = ({ isFiltered }) => (
...
+    { content: 'Device status', key: 'status', Component: ({ deviceDeployment: { status } }) => status },
+    {
+      content: '',
+      key: 'log',
+      Component: ({ deviceDeployment: { id, log, deviceId } }) =>
+        log && (
+          <Button
+            component="a"
+            href={`${window.location.origin}${deploymentsApiUrl}/deployments/${id}/devices/${deviceId}/log`}
+            onClick={() =>
```
Comment (aleksandrychev) [inline src/js/components/devices/device-details/deployments.js:93]: there is createDownload function in the helpers.js that makes cookie set

```
export const createDownload = (target, filename, token) => {
  let link = document.createElement('a');
  link.setAttribute('href', target);
  link.setAttribute('download', filename);
  link.style.display = 'none';
  document.body.appendChild(link);
  cookies.set('JWT', token, { path: '/', secure: true, sameSite: 'strict', maxAge: 5 });
  link.click();
  document.body.removeChild(link);
};
```

**Review by aleksandrychev — CHANGES_REQUESTED** [review]

**Comment by aleksandrychev** [comment]
@mineralsfree please update snapshot and then good to go

**Review by aleksandrychev — APPROVED** [review]

**Comment by aleksandrychev** [comment]
@tranchitella backport?

**Comment by tranchitella** [comment]
> @tranchitella backport?

no need to backport this fix.

**Review by tranchitella — APPROVED** [review]

**Pull request merged**